src/main/java/org/intellij/sdk/codesync/codeSyncSetup/CodeSyncSetup.java

```diff
@@ -269,6 +272,8 @@ public static boolean checkUserAccess(Project project, String repoPath, String r
                 return validateAccessToken(accessToken);
             } catch (InvalidAccessTokenError invalidAccessTokenError) {
```

Comment:
Are we catching 2 same exceptions here? @syedgulahmed

Reply:
@Basit-qc Removed it.

Reply:
@syedgulahmed I am seeing following
![image](https://user-images.githubusercontent.com/9863016/235434113-6f678174-8033-4484-926a-b7ed5710662a.png)

What did you remove exactly?

Reply:
I mean I removed it from catch block. Also changed the implementation, now state value is being changed according to return value of validateAccessToken method.

Reply:
Any idea why is it still visible in the Catch Block? I don't it being removed in the PR changes, attached screenshot as well

Reply:
![image](https://user-images.githubusercontent.com/34773162/235439020-72a968f4-7582-43a2-986c-be2edd69ada2.png)

Change the commit from here 
![image](https://user-images.githubusercontent.com/34773162/235439071-bf2d0b50-bca5-4df6-8a52-4d3eea10919d.png)

Reply:
@Basit-qc

Reply:
That seems invalid change @syedgulahmed 
![image](https://user-images.githubusercontent.com/9863016/235439275-21dd3c11-c726-4f55-a35b-41f3039cb439.png)
Did you test it?

Reply:
@syedgulahmed I do not think you need this change, the code https://github.com/codesyncapp/codesync-intellij/pull/89/files#diff-64d5482f6cda56550ce5dcf5149ba7a2ba2a31d9ab775eb6a25b56ba4ce67477R143 will execute regardless of the outcome of this logic. I think the only changes required in this PR are https://github.com/codesyncapp/codesync-intellij/pull/89/files#diff-64d5482f6cda56550ce5dcf5149ba7a2ba2a31d9ab775eb6a25b56ba4ce67477R128-R143 

However, there are certain cases with async flow where you might need to make more changes.

My suggestion would to start with just https://github.com/codesyncapp/codesync-intellij/pull/89/files#diff-64d5482f6cda56550ce5dcf5149ba7a2ba2a31d9ab775eb6a25b56ba4ce67477R128-R143 changes and only make changes in other places where needed.